sessionctx/variable/session.go

```diff
@@ -112,6 +112,8 @@ type SessionVars struct {
 	PreparedStmtNameToID map[string]uint32
 	// preparedStmtID is id of prepared statement.
 	preparedStmtID uint32
+	// params for prepared statements
+	Params []interface{}
```

Comment:
Thank you for the sugguestions. 

(1) I agree your idea to store the parameter values as expressions instread of interfaces. However, since I could not resolve cycle dependency issues easily when replacing the interfaces with expressions, I'd like to handle it as an independant issue. 

(2) I'v changed the function (i.e., getparam) returning the values of types.Datum based on the new expression evaluation framework. Currently, since the value type of parameters is always VarString, the string-typed function is only supported. In order to support more typed functions, we need to also support "typed parameters".